compiler/frontend/beam/src/main/java/org/apache/nemo/compiler/frontend/beam/NemoPipelineResult.java

```diff
@@ -24,17 +24,28 @@
 import org.joda.time.Duration;
 
 import java.io.IOException;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
 
 /**
  * Beam result.
  */
 public final class NemoPipelineResult extends ClientEndpoint implements PipelineResult {
+  private final CountDownLatch jobDone;
 
   /**
    * Default constructor.
    */
   public NemoPipelineResult() {
     super(new BeamStateTranslator());
+    jobDone = new CountDownLatch(1);
```

Comment:
Why don't you get this `CountDownLatch` as an argument? `NemoRunner` can inject it to this, and you can remove `triggerJobDone()`

Reply:
I think that's just a matter of style: whether you feel more comfortable with dealing with the latch yourself or with simply calling the method. What do you think is the advantage over the current implementation?